gips/data/core.py

```diff
@@ -563,18 +574,21 @@ def _archivefile(cls, filename, update=False):
                 # check if another asset exists
                 existing = cls.discover(asset.tile, d, asset.asset)
                 if len(existing) > 0 and (not update or not existing[0].updated(asset)):
-                    # gatekeeper case:  No action taken because existing assets are in the way
+                    # gatekeeper case:  No action taken because other assets exist
                     VerboseOut('%s: other version(s) already exists:' % bname, 1)
                     for ef in existing:
                         VerboseOut('\t%s' % os.path.basename(ef.filename), 1)
                     otherversions = True
                 elif len(existing) > 0 and update:
-                    # update case:  Remove existing outdated assets and install the new one
+                    # update case:  Remove existing outdated assets
+                    #               and install the new one
                     VerboseOut('%s: removing other version(s):' % bname, 1)
                     for ef in existing:
-                        assert ef.updated(asset), 'Asset is not updated version'
+                        if not ef.updated(asset):
+                            'Asset is not updated version'
```

Comment:
Is this line missing a call or an assignment to go with the string?

Reply:
Now that is getting distracted at an inopportune, but syntactically correct time.  Was going to replace the assertionerror, but hadn't figured it, and got pulled into something else.